src/compiler/utilities.ts

```diff
@@ -215,8 +215,10 @@ namespace ts {
 
     // Gets the nearest enclosing block scope container that has the provided node
     // as a descendant, that is not the provided node.
-    export function getEnclosingBlockScopeContainer(node: Node): Node {
-        let current = node.parent;
+    export function getEnclosingBlockScopeContainer(node: Node): Node;
+    export function getEnclosingBlockScopeContainer(node: Node, getAncestorOrSelf: (offset: number) => Node, offset: number): Node;
+    export function getEnclosingBlockScopeContainer(node: Node, getAncestorOrSelf?: (offset: number) => Node, offset?: number): Node {
+        let current = getAncestorOrSelf ? getAncestorOrSelf(++offset) : node.parent;

```

Comment:
Just looking at it, I'm not sure what the implied behavior of `getAncestorOfSelf` is doing with `offset` and why you need to increment it when feeding the value. A (potentially JSDoc) comment would be helpful.